cmd/helm/plugin_install.go

```diff
@@ -83,10 +84,20 @@ func (pcmd *pluginInstallCmd) run() error {
 		return err
 	}
 
+	// Make sure a command with this name does not already exist.
+	pluginCmdName := p.Metadata.Name
+	for _, cmd := range RootCmd.Commands() {
+		if cmd.Name() == pluginCmdName {
```

Comment:
I wonder can this be tested earlier in the flow? It's always better to check first rather than adding and then removing.

Reply:
@hickeyma Thank you very much! I couldn't agree with you more. However, I can not get the plugin information without installing it locally. And If we use a temporary directory first, there is also the problem of repeated installations. Cleaning up after a bad installation may be easier and faster than reinstalling. If you know how to get the plugin information without installing it, please let me know. Looking forward to your reply!